RampUpAuto/src/GithubProject/TCs_GitHubLoginProcess.java

```diff
@@ -0,0 +1,75 @@
+package RampUpAuto.src.GithubProject;
+
+import org.testng.annotations.*;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.edge.EdgeDriver;
+import org.openqa.selenium.firefox.FirefoxDriver;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Parameters;
+
+
+import java.lang.reflect.Method;
+
+public class TCs_GitHubLoginProcess {
+
+    public WebDriver Driver = null;
+
+    @DataProvider(name = "Test")
+    public Object[][] githubSigninDb(Method m) {
+        switch (m.getName()) {
+            case "validSignInProcess":
+                return new Object[][]{
+                        {"asaltest19@gmail.com", "password123456789s*"},
+                };
+
+            case "invalidSignInProcess":
+                return new Object[][]{
+                        {"asaltest1646449@gmail.com", "password123456789s*"},
+                        {"asaltest19@gmail.com", "password123456789sgg"},
+                        {"", ""}
+                };
+        }
+        return null;
+
+    }
+
+    ;
+
+
+    @BeforeMethod
+    @Parameters("browser")
+    public void setup(String browser) {
+        if (browser.equalsIgnoreCase("chrome")) {
+            System.setProperty("webdriver.chrome.driver", "/Users/sajasalahat/Desktop/chromedriver-mac-x64");
+            Driver = new ChromeDriver();
+            Driver.get("https://github.com/login");
+        } else if (browser.equalsIgnoreCase("edge")) {
+            System.setProperty("webdriver.edge.driver", "/Users/sajasalahat/Desktop/msedgedriver 3");
+            Driver = new EdgeDriver();
+            Driver.get("https://github.com/login");
+
+        } else
+            throw new IllegalArgumentException("Invalid");
+
+    }
+
+    @Test(dataProvider = "Test", priority = 1)
+    public void validSignInProcess(String Email, String Password) {
+
+        GitHubSignIn GitHubSignIn_ = new GitHubSignIn();
+        GitHubSignIn_.LoginIndicator(Driver, Email, Password);
+        GitHubSignIn_.clickSigninButton(Driver);
+
+    }
+
+    @Test(dataProvider = "Test", priority = 2)
```

Comment:
instead of creating 2 separate functions , you can merge them into just one function , so this way we will eliminate the duplication in the code and enhance the maintainability and reusability of it 
for example , you can add these lines of codes into one single function 
` GitHubSignIn GitHubSignIn_ = new GitHubSignIn();
        GitHubSignIn_.LoginIndicator(Driver, Email, Password);
        GitHubSignIn_.clickSigninButton(Driver);`
and you can add like flag variables (true or false ) to your `githubSigninDb` as this         ```
  {"asaltest19@gmail.com", "password123456789s*, true"},
                };
```
`                        {"asaltest19@gmail.com", "password123456789sgg",false},`
and in this function you can add if statement to check for the valid and the invalid cases